### PR TITLE
Production Deployment

### LINKED ISSUE #39: Improve dialog design
The current dialog design seem a bit outdated and could use a new polish.

### LINKED ISSUE #40: Add custom dialogs
At the moment all dialogs have the same design. However, it would be nicer (*especially for the export format selection*) to design the content of the dialog itself.

### Concept
I picture something like this:

image

Maybe this is possible with [content projection](https://angular.io/guide/content-projection#single-slot-content-projection)?

### Depends on
- #39

### LINKED ISSUE #72: [Feature]: New name for duplicate palettes
### Feature Description

Instead of naming a copy of a palette `<name> (copy)`, we can instead prompt the user to enter a new name for their copy.

### Additional information

Follow-up to:
- #63 

Depends on:
- #71

### LINKED ISSUE #73: [Bug]: Hide warning about unsaved changes during upgrade
### What happened?

After confirming to update the app live when the new version was downloaded I was still prompted about my unsaved changes

### Additional Notes

Solution: Disable the check for unsaved changes while doing the live update.

### PR DESCRIPTION
**Pull request by pawcoding**
### New Features ✨ 
- Re-order colors with Drag-and-Drop
- Add animation when no-palette is available
- Custom dialogs for confirm actions and renaming
- Ask user for new name when duplicating palette
- Add confetti service
  - Shoot confetti after successful export
  - Shoot confetti after successful update

### Bug fixes 🩹 
- Better error handling when clipboard is not available
- Don't warn about unsaved changes during update
- Disable export button when palette has no colors
- Update German translation for renamed color toast

### Internal 🛠️ 
- Add support for asynchronous palette generation
- Add support for custom toast duration (even permanent toasts)
- Simplify internals of dropdown menu
- Update dependencies

## Includes
- #58 (nice try, but didn't work)
- Closes #73 
- #70 
- #71 
  - Closes #39 
  - Closes #40
- #76 
  - Closes #72
- #77

### CODE REVIEW

File: src/app/shared/ui/color-input/color-input.component.html [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
       [title]="tooltip()"
-      for="color"
     >
       <div
-        [class]="isColorLight() ? 'text-neutral-900' : 'text-neutral-100'"
+        [class]="
+          (isColorLight() ? 'text-neutral-900' : 'text-neutral-100') +
+          ' ' +
+          (readOnly() ? 'cursor-not-allowed' : 'cursor-pointer')
+        "
```
Comment (kevinknecht-ai) [inline src/app/shared/ui/color-input/color-input.component.html]: Maybe something like that (to take some logic and concatenation out of the template)

```suggestion
       [class]="iconStyle()"
```
```

 protected iconStyle = computed(() => {
    const colorLight = this.isColorLight() ? 'text-neutral-900' : 'text-neutral-100';
    const readOnly = this.readOnly() ? 'cursor-not-allowed' : 'cursor-pointer';
    return [colorLight, readOnly].join(' ');
  })

```

**Review by kevinknecht-ai — COMMENTED** [review]
Had a quick look at this one (out of interest). Looks very good!

File: src/app/shared/ui/color-input/color-input.component.html (hunk 1 above)
Comment (pawcoding) [reply src/app/shared/ui/color-input/color-input.component.html]: IMO it's always a trade off. I see why your suggestion is nicer to read (that's why I implemented it), but now you have to take a look at TS code to see the full style of the component.

**Pull request merged**